pkg/operator/csi/csidefaultstorageclasscontroller/csi_default_storageclass_controller.go

```diff
@@ -0,0 +1,135 @@
+package csidefaultstorageclasscontroller
+
+import (
+	"context"
+	operatorapi "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/listers/storage/v1"
+	"k8s.io/klog/v2"
+	"time"
+)
+
+const (
+	conditionsPrefix       = "CSIDefaultStorageClassController"
+	defaultScAnnotationKey = "storageclass.kubernetes.io/is-default-class"
+)
+
+// This Controller deploys a StorageClass provided by CSI driver operator
+// and decides if this StorageClass should be applied as default - if requested.
+// If operator wants to request it's StorageClass to be created as default,
+// the asset file provided to this controller must have defaultScAnnotationKey set to "true".
+// Based on the current StorageClasses in the cluster the controller can decide not
+// to deploy given StorageClass as default if there is already an existing default.
+// Also, the controller will emit alerts if more than one StorageClass defined
+// as default exists - this is not a recommended setup.
```

Comment:
Please remove this, it's for a subsequent PR.

Reply:
done